core/src/replay_stage.rs

```diff
@@ -3876,7 +3926,25 @@ impl ReplayStage {
                 if bank.is_frozen() && tower.is_slot_confirmed(*slot, voted_stakes, total_stake) {
                     info!("validator fork confirmed {} {}ms", *slot, duration);
                     datapoint_info!("validator-confirmation", ("duration_ms", duration, i64));
-                    confirmed_forks.push((*slot, bank.hash()));
+                    confirmed_forks.push(ConfirmedSlot::new_optimistic_confirmed_slot(
+                        *slot,
+                        bank.hash(),
+                    ));
+                } else if bank.is_frozen()
+                    && tower.is_slot_duplicate_confirmed(*slot, voted_stakes, total_stake)
```

Comment:
previously we were only calling `mark_slots_confirmed` if we got `tower.is_slot_confirmed` . this change allows us to notify the fsm as soon as we have `tower.is_slot_duplicate_confirmed`